### PR TITLE
coverage on golang integration tests

### PR DESCRIPTION
**Pull request by shay6**

### CODE REVIEW

File: documentation/golang/README.md [hunk 1]
```diff
@@ -677,6 +677,270 @@ published coverage report dashboard such as this one:
          }
       }
 
+
+# Code Coverage with integration tests
```
Comment (simzacks) [inline documentation/golang/README.md:681]: This reads like you copied it from somewhere, if so, can you add the URL of where you copied it from as a reference?

File: documentation/golang/README.md [hunk 2]
```diff
@@ -677,6 +677,270 @@ published coverage report dashboard such as this one:
...
+
+Additionally, you can get this special “test binary” if you pass -c -o <file> to go test, allowing you to run the tests separately (without recompiling the source code each time). Note that the test binary will only include the tests in the package that you explicitly state in the package line (imported packages won’t have their tests run implicitly).
+
+As explained above, when you pass -cover to go test the compiler will add a bunch of instrumentation to your source code to count how often a line of code was executed. So if we want coverage on integration tests we will have to recompile the source code with cover flags. Unfortunately it’s not (easily) possible to get this cover tool to generate such instrumentation using "go build". So if you want code coverage you need to use "go test".
```
Comment (simzacks) [inline documentation/golang/README.md]: > when you pass -cover to go test
add a comma (,) after go test
also maybe add quotes around  "go test" in every instance it is used.

File: documentation/golang/README.md (hunk 1 above)
Comment (shay6) [reply documentation/golang/README.md:681]: done

File: documentation/golang/README.md (hunk 2 above)
Comment (shay6) [reply documentation/golang/README.md]: done

**Pull request merged**